src/main/java/com/work/management/service/DataPersistingService.java

```diff
@@ -0,0 +1,14 @@
+package com.work.management.service;
+
+public interface DataPersistingService<T,V> {
+
+    public T preHandle(T document);
+
```

Comment:
In a interface everything  is by default public... you don't need to add public explicitly here..its just creating noise...remove public from all methods in this interface.